deployments/mantis/goguen-ala-cardano-target-aws.nix

```diff
@@ -0,0 +1,71 @@
+with builtins;
+with import ../../lib.nix;
+{ accessKeyId, deployerIP, ... }:
+let
+    org             = "IOHK";
+    region          = "eu-central-1";
+
+    sgByName        = resources: org: region: name:
+      resources.ec2SecurityGroups."${name}-${region}-${org}";
+
+    mkMachine = sgs: { resources, ... }: {
+      imports = [ <module/amazon-base.nix> ];
+      deployment.ec2 = {
+        securityGroups = map (sgByName resources org region) (["allow-deployer-ssh"] ++ sgs);
+        blockDeviceMapping = {
+          "/dev/xvdf" = { size = 120; }; # resources.ebsVolumes."${hostname}-ebs"
+        };
+        associatePublicIpAddress = true;
+      };
+    };
+    mkExplorer = mkMachine ["allow-explorer-public"];
+    mkFaucet   = mkMachine ["allow-faucet-public"];
+    mkMantis   = mkMachine ["allow-mantis-public"];
+in {
+  explorer-a = mkExplorer;
+
+  faucet-a   = mkFaucet;
+
+  mantis-a-0 = mkMantis;
+  mantis-a-1 = mkMantis;
+  mantis-b-0 = mkMantis;
+  mantis-b-1 = mkMantis;
+  mantis-c-0 = mkMantis;
+
+  resources = {
+    elasticIPs = {
+      explorer-a-ip = { inherit region accessKeyId; };
+
+      faucet-a-ip   = { inherit region accessKeyId; };
+
+      mantis-a-0-ip = { inherit region accessKeyId; };
+      mantis-a-1-ip = { inherit region accessKeyId; };
+      mantis-b-0-ip = { inherit region accessKeyId; };
+      mantis-b-1-ip = { inherit region accessKeyId; };
+      mantis-c-0-ip = { inherit region accessKeyId; };
+    };
+
+    ec2SecurityGroups =
+    let public = type: port: { protocol = type; fromPort = port ; toPort = port ; sourceIp = "0.0.0.0/0"; };
```

Comment:
minor nitpick, if you change the function to `protocol: port:` then you can just `{ inherit protocol; ...` instead of `protocol = type;`

and then port as well, `protocol: toPort:` and `{ inherit protocol toPort; ...`

Reply:
I'm looking towards removing the initial G-ala-C implementation at some point in the future.
We can't do it for a reason right now.
In this light, I'm not sure it'd be an economic expenditure of effort to focus on making this pretty.